### PR TITLE
Add network policy

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/4033

### CODE REVIEW

**Comment by corest** [comment]
@giantswarm/team-batman PTAL

File: helm/kubernetes-nginx-ingress-controller-chart/charts/http-backend-service-type-migration/templates/np.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: {{ .Values.name }}-allow
+  namespace: {{ .Values.namespace }}
+  annotations:
```
Comment (pipo02mix) [inline helm/kubernetes-nginx-ingress-controller-chart/charts/http-backend-service-type-migration/templates/np.yaml]:
```suggestion
```

File: helm/kubernetes-nginx-ingress-controller-chart/charts/cleanup/templates/np.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: {{ .Values.name }}-allow
+  namespace: {{ .Values.namespace }}
+  annotations:
+  annotations:
```
Comment (pipo02mix) [inline helm/kubernetes-nginx-ingress-controller-chart/charts/cleanup/templates/np.yaml]:
```suggestion
```

File: helm/kubernetes-nginx-ingress-controller-chart/charts/migration/templates/np.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: {{ .Values.name }}-allow
+  namespace: {{ .Values.namespace }}
+  annotations:
+  annotations:
```
Comment (pipo02mix) [inline helm/kubernetes-nginx-ingress-controller-chart/charts/migration/templates/np.yaml]:
```suggestion
```

File: helm/kubernetes-nginx-ingress-controller-chart/charts/tempresources/templates/np.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: {{ .Values.name }}-allow
+  namespace: {{ .Values.namespace }}
+  annotations:
+  annotations:
```
Comment (pipo02mix) [inline helm/kubernetes-nginx-ingress-controller-chart/charts/tempresources/templates/np.yaml]:
```suggestion
```

File: helm/kubernetes-nginx-ingress-controller-chart/charts/http-backend-service-type-migration/templates/np.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: {{ .Values.name }}-allow
```
Comment (pipo02mix) [inline helm/kubernetes-nginx-ingress-controller-chart/charts/http-backend-service-type-migration/templates/np.yaml]: It has same name/namespace than np before and following. there will be a collision?

**Review by pipo02mix — CHANGES_REQUESTED** [review]
some dups

File: helm/kubernetes-nginx-ingress-controller-chart/charts/http-backend-service-type-migration/templates/np.yaml (hunk 5 above)
Comment (corest) [reply helm/kubernetes-nginx-ingress-controller-chart/charts/http-backend-service-type-migration/templates/np.yaml]: Migrations have their own names

Comment (pipo02mix) [reply helm/kubernetes-nginx-ingress-controller-chart/charts/http-backend-service-type-migration/templates/np.yaml]: fair point 🤦‍♂

**Comment by corest** [comment]
@pipo02mix fixed

**Review by pipo02mix — APPROVED** [review]
LGTM

**Pull request merged**